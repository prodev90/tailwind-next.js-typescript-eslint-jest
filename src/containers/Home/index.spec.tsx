import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'

import Index from './index'

describe('Index Page', () => {
  let uut: ShallowWrapper

  beforeEach(() => {
    uut = shallow(<Index />)
  })

  it('renders correctly', () => {
    expect(uut).toMatchSnapshot()
  })

  it('renders hello', () => {
    // Given
    const EXPECTED = 'Hello'

    // When
    const ACTUAL = uut.find('h1').text()

    // Then
    expect(ACTUAL).toEqual(EXPECTED)
  })

  it('renders hello with tailwind classes', () => {
    // Given
    const EXPECTED = 'text-xl text-gray-900'

    // When
    const ACTUAL = uut.find('h1').props().className

    // Then
    expect(ACTUAL).toEqual(EXPECTED)
  })
})
