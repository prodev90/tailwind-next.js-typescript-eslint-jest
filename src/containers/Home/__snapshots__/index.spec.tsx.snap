// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Index Page renders correctly 1'] = `
<h1
  className="text-xl text-gray-900"
>
  Hello
</h1>
`
